code/messaging/src/main/java/com/adobe/marketing/mobile/MessagingPushPayload.java

```diff
@@ -108,6 +109,12 @@ public MessagingPushPayload(final RemoteMessage message) {
             Log.error(LOG_TAG, SELF_TAG, "Failed to create MessagingPushPayload, remote message data payload is null");
             return;
         }
+
+        if(StringUtils.isNullOrEmpty(message.getMessageId())) {
+            Log.error(LOG_TAG, SELF_TAG, "Failed to create MessagingPushPayload, message id is null or empty");
+            return;
+        }
+        messageId = message.getMessageId();
```

Comment:
nit: get the message id before the if statement on line 113 so there is only one call to getMessageId()